src/Simulation/Common/PortablePDBReader.cs

```diff
@@ -0,0 +1,396 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Quantum.Simulation.Common
+{
+    using System.Collections.Generic;
+    using System.Reflection.Metadata;
+    using System;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Text;
+    using Newtonsoft.Json.Linq;
+    using System.Text.RegularExpressions;
+    using Microsoft.Quantum.Simulation.Core;
+    using System.Linq;
+
+    public class CompressedSourceFile
+    {
+        private readonly byte[] compressedSource = null;
+        private string decompressedSource = null;
+
+        public CompressedSourceFile(byte[] compressedSource)
+        {
+            this.compressedSource = compressedSource;
+        }
+
+        public override string ToString()
+        {
+            if (decompressedSource == null)
+            {
+                using MemoryStream memoryStream = new MemoryStream(compressedSource, sizeof(Int32), compressedSource.Length - sizeof(Int32));
+                using DeflateStream decompressionStream = new DeflateStream(memoryStream, CompressionMode.Decompress);
+                Int32 uncompressedSourceFileSize = BitConverter.ToInt32(compressedSource, 0);
+                MemoryStream decompressed = new MemoryStream(new byte[uncompressedSourceFileSize], true);
+                decompressionStream.CopyTo(decompressed);
+                decompressedSource = Encoding.UTF8.GetString(decompressed.ToArray());
+            }
+
+            return decompressedSource;
+        }
+    }
+
+    /// <summary>
+    /// Class for source link information corresponding to SourceLink schema 
+    /// https://github.com/dotnet/designs/blob/master/accepted/diagnostics/source-link.md#source-link-json-schema
+    /// </summary>
+    [Serializable]
+    public class SourceLinkPathRemapping
```

Comment:
Wondering if this should just be called `SourceLink` or `SourceLinkData` or `SourceLinkInfo`.